### PR TITLE
Fix positive raw scores for TreeEnsembleClassifier

### PR DESCRIPTION
**Pull request by xadupre**
**Description**:

Raw scores for TreeEnsembleClassifier (transform is NONE) always have a first negative column and a second positive column for a binary classification problem.

**Motivation and Context**

Raw scores are wrong for model HistGradientBoostingClassifier.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/ml/ml_common.h [hunk 1]
```diff
@@ -330,6 +331,7 @@ void write_scores(std::vector<T>& scores, POST_EVAL_TRANSFORM post_transform, in
             scores[0] = -scores[0];
           }
           break;
+        /*
         case 3:  //3 = mixed weights, winning class is negative
```
Comment (prabhat00155) [inline onnxruntime/core/providers/cpu/ml/ml_common.h]: Why do we have commented out code here?

File: onnxruntime/core/providers/cpu/ml/ml_common.h [hunk 2]
```diff
@@ -322,6 +322,7 @@ void write_scores(std::vector<T>& scores, POST_EVAL_TRANSFORM post_transform, in
           scores[0] = 1.f - scores[0];
           break;
         case 2:  //2 = mixed weights, winning class is positive
+        case 3:  //3 = mixed weights, winning class is negative
           if (post_transform == POST_EVAL_TRANSFORM::LOGISTIC) {
             scores.push_back(ComputeLogistic(scores[0]));  //ml_logit(scores[k]);
             scores[0] = ComputeLogistic(-scores[0]);
```
Comment (prabhat00155) [inline onnxruntime/core/providers/cpu/ml/ml_common.h]: These two could be just one line: 
scores[0] = -1 * ComputeLogistic(-scores[0]);

File: onnxruntime/core/providers/cpu/ml/ml_common.h (hunk 1 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/ml_common.h]: Fixed

File: onnxruntime/core/providers/cpu/ml/ml_common.h (hunk 2 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/ml_common.h]: I prefer to leave it that way. It may introduce minor discrepencies.

**Pull request merged**